dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -123,4 +127,126 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
     {
         return AppModel::Identity::IsPackagedProcess() && IsBackgroundTaskBuilderAvailable();
     }
+
+    inline HRESULT GetPackageFullName(wil::unique_cotaskmem_string& packagedFullName) noexcept try
+    {
+        WCHAR packageFullName[PACKAGE_FULL_NAME_MAX_LENGTH + 1]{};
+        UINT32 packageFullNameLength = ARRAYSIZE(packageFullName);
+        THROW_IF_FAILED(GetCurrentPackageFullName(&packageFullNameLength, packageFullName));
+
+        packagedFullName = wil::make_cotaskmem_string_nothrow(packageFullName);
+        THROW_IF_NULL_ALLOC(packagedFullName);
+
+        return S_OK;
+    }
+    CATCH_RETURN()
+
+    inline HRESULT GetExeNameFromPath(std::wstring& exePath) noexcept try
+    {
+        size_t pos{ exePath.rfind(L"\\") };
+        THROW_HR_IF(E_UNEXPECTED, pos == std::wstring::npos);
+        exePath = exePath.substr(pos + 1); // One after the delimiter
+
+        return S_OK;
+    }
+    CATCH_RETURN()
+
+    inline HRESULT CheckComServerClsid(winrt::guid& comServerClsid, const std::wstring argumentToCheck) noexcept try
+    {
+        wil::unique_cotaskmem_string packagedFullName;
+        THROW_IF_FAILED(GetPackageFullName(packagedFullName));
+
+        wil::unique_hkey hKey;
+        // Path where packaged apps' ComActivators are in the registry:
+        // HKLM\SOFTWARE\Classes\PackagedCom\Package\{PackageFullName}\Class
+        std::wstring packagedRegistryPath{ LR"(SOFTWARE\Classes\PackagedCom\Package\)" + std::wstring(packagedFullName.get()) };
+        std::wstring clsidPath{ packagedRegistryPath + LR"(\Class)"};
+        THROW_IF_FAILED(RegOpenKeyEx(HKEY_LOCAL_MACHINE, clsidPath.c_str(), 0, KEY_READ, &hKey));
+
+        DWORD comServerGuids{};
+        DWORD maxSubKeyLen;
+        THROW_IF_FAILED(::RegQueryInfoKey(
+            hKey.get(),
+            nullptr,    // No user-defined class
+            nullptr,    // No user-defined class size
+            nullptr,    // Reserved
+            &comServerGuids,
+            &maxSubKeyLen,
+            nullptr,    // No subkey class length
+            nullptr, // No values we want to get
+            nullptr,
+            nullptr,    // No max value length
+            nullptr,    // No security descriptor
+            nullptr     // No last write time
+        ));
+
+        maxSubKeyLen++; // Account for the null character
+
```

Comment:
THROW_HR_IF_MSG(E_FAIL, comServerGuidCount < 1, "No COM servers are registered for this app);